dcmtk: new version and pic configuration

aumuell opened the pull request:
pic variant is required if the DICOM I/O support plugin for OpenSceneGraph should be built

aumuell committed:
dcmtk: checksum 3.6.7

compiles on macos

aumuell committed:
dcmtk: support pic configuration

aumuell commented:
@spackbot fix style

aumuell committed:
[@spackbot] updating style on behalf of aumuell

alecbcs commented on var/spack/repos/builtin/packages/dcmtk/package.py:
```diff
@@ -41,9 +42,12 @@ class Dcmtk(CMakePackage):
     variant("cxx11", default=False, description="Enable c++11 features")
     variant("stl", default=True, description="Use native STL implementation")
 
+    conflicts("platform=darwin", when="@:3.6.6")
```
@aumuell Is there any particular reason why we're setting a conflicts with MacOS here? In the [`ANNOUNCE` file for v3.6.5](https://nero.offis.de/download/dcmtk/dcmtk365/ANNOUNCE) it seems MacOS is stated as a supported target.

aumuell replied:
Thank you: in deed, looking into the error message showed that the problem only exists on macos and aarch64. I changed the conflicts accordingly

tldahlgren reviewed (commented):
Confirmed the version sha256.

While you're matching the current style for setting `cmake_args`, there are helpers that could be used (see https://spack.readthedocs.io/en/latest/build_systems/cmakepackage.html#adding-flags-to-cmake).

aumuell committed:
dcmtk: use define_from_variant for shorter code

aumuell committed:
dcmtk: refine conflict

it appears that dcmtk < 3.6.7 only fails on macos/aarch64:
/Library/Developer/CommandLineTools/usr/lib/clang/14.0.3/include/xmmintrin.h:14:2:
  error: "This header is only meant to be used on x86 and x64 architecture"

aumuell commented:
> Confirmed the version sha256.
> 
> While you're matching the current style for setting `cmake_args`, there are helpers that could be used (see https://spack.readthedocs.io/en/latest/build_systems/cmakepackage.html#adding-flags-to-cmake).

Thank you for the hint: I changed all cmake_args to use `define_from_variant`.

tldahlgren reviewed (approved):
Builds successfully for me and LGTM.

alecbcs reviewed (approved):
Looks good to me. Thanks @aumuell for the new variant and retrofitting the args statements.
